### ISSUE TITLE
New Rule Idea: Attribute based exclusion for Integration testable code

### ISSUE THREAD

**Issue by Hefaistos68** [SonarSource/sonar-dotnet#9788]
### Description

Some code can only be tested through integration tests, reporting them as not covered is often an issue. 
Flagging those with a [IntegrationTestableOnly] attribute or similar could solve this issue, especially when qualitygates are defined with coverage rules.

### Noncompliant code snippet

```
private static void ConfigureServices(IServiceCollection services, IConfiguration config)
{
    // Register the Config as a singleton.
    services.AddSingleton(m => config);   
}
```

### Compliant code snippet

```
[IntegrationTestableOnly]
private static void ConfigureServices(IServiceCollection services, IConfiguration config)
{
    // Register the Config as a singleton.
    services.AddSingleton(m => config);   
}
```

**Comment by Hefaistos68** [comment]
Of course, this applies only if you DONT have integration tests, otherwise the test project rules would apply.

**Comment by pavel-mikula-sonarsource** [comment]
Hi,

this isn't a rule idea, as it does not raise an issue, but a feature of your coverage tool.

Coverage tool(s) typically have a way how to configure an attribute that makes your coverage tool not report coverage for certain method. Then, you won't import coverage to SonarQube and you won't see those.

For example, in this repo, we use the built-in `ExcludeFromCodeCoverage` attribute and configure our coverage tool to respect that (see the [yml in this repo](https://github.com/SonarSource/sonar-dotnet/blob/master/azure-pipelines.yml)).
